confgenerator/confmerger.go

```diff
@@ -0,0 +1,150 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package confgenerator provides functions to generate subagents configuration from unified agent.
+package confgenerator
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/viper"
+)
+
+var (
+	builtInConf = map[string]string{
+		"linux": `logging:
+  receivers:
+    syslog:
+      type: files
+      include_paths:
+      - /var/log/messages
+      - /var/log/syslog
+  exporters:
+    google:
+      type: google_cloud_logging
+  service:
+    pipelines:
+      default_pipeline:
+        receivers: [syslog]
+        exporters: [google]
+
+metrics:
+  receivers:
+    hostmetrics:
+      type: hostmetrics
+      collection_interval: 60s
+  exporters:
+    google:
+      type: google_cloud_monitoring
+  service:
+    pipelines:
+      default_pipeline:
+        receivers: [hostmetrics]
+        exporters: [google]`,
+
+		"windows": `logging:
+  receivers:
+    windows_event_log:
+      type: windows_event_log
+      channels: [System,Application,Security]
+  exporters:
+    google:
+      type: google_cloud_logging
+  service:
+    pipelines:
+      default_pipeline:
+        receivers: [windows_event_log]
+        exporters: [google]
+
+metrics:
+  receivers:
+    hostmetrics:
+      type: hostmetrics
+      collection_interval: 60s
+    mssql:
+      type: mssql
+      collection_interval: 60s
+    iis:
+      type: iis
+      collection_interval: 60s
+  exporters:
+    google:
+      type: google_cloud_monitoring
+  service:
+    pipelines:
+      default_pipeline:
+        receivers: [hostmetrics,mssql,iis]
+        exporters: [google]`,
+	}
+)
+
+func MergeConfFiles(builtInConfPath, userConfPath, mergedConfPath, platform string) error {
+	builtInConfFolder := path.Dir(builtInConfPath)
+	builtInConfFullName := path.Base(builtInConfPath)
+	builtInConfName := strings.TrimSuffix(builtInConfFullName, filepath.Ext(builtInConfFullName))
+	userConfFolder := path.Dir(userConfPath)
+	userConfFullName := path.Base(userConfPath)
+	userConfName := strings.TrimSuffix(userConfFullName, filepath.Ext(userConfFullName))
+	mergedConfFolder := path.Dir(mergedConfPath)
+
+	// Write the built-in conf to disk for debugging purpose.
+	f, err := os.Create(builtInConfPath)
+	if err != nil {
+		return fmt.Errorf("error creating the built-in config file %q: %w \n", builtInConfPath, err)
+	}
+	defer f.Close()
+	if _, err = f.WriteString(builtInConf[platform]); err != nil {
+		return fmt.Errorf("error writing the built-in config file %q: %w \n", builtInConfPath, err)
+	}
+
+	// Init Viper to process YAML format configs.
+	viper := viper.New()
+	viper.SetConfigType("yaml")
+
+	// Read the built-in config file.
+	viper.AddConfigPath(builtInConfFolder)
+	viper.SetConfigName(builtInConfName)
+	err = viper.ReadInConfig()
+	if err != nil {
```

Comment:
Go style is to do this on one line.

    if err = viper.ReadInConfig(); err != nil {
      ...
    }

Reply:
Changed.